File: geometry/polar_cartesian_conversion/python/polar_and_cartesian.py
import math
from math import pi, sin, cos

def polar_to_cartesian(radius, degrees):
  r = float(radius)
  d = float(degrees)

  theta = d*pi/180

  x = r*cos(theta)
  y = r*sin(theta)
  return x,y
  
def cartesian_to_polar(x, y):
  x = float(x)
  y = float(y)

  r = math.sqrt(x*x + y*y)
  theta = math.atan2(y, x)
  return r, theta

File: geometry/polar_cartesian_conversion/java/PolarAndCartesian.java
import java.util.*;

class PolarAndCartesian{

    public static double[] cartesianToPolar(double x, double y){
    	
		double[] rAndTheta = new double[2];
    	rAndTheta[0] = Math.sqrt(x*x + y*y); //radius
    	rAndTheta[1] = Math.atan2(y, x); 	 //theta
    	return rAndTheta;
    }

    public static double[] polarToCartesian(double r, double theta){
    	double[] cartesianCoords = new double[2];
    	cartesianCoords[0] = Math.cos(theta)*r; //x value
    	cartesianCoords[1] = Math.sin(theta)*r; //y value
    	return cartesianCoords;


    }
}
